### ISSUE TITLE
Coverage report is not showing 100% on the second run

### ISSUE THREAD

**Issue by winksaville** [alsatian-test/alsatian#377]
The first time I do a `npm run test` after doing a `npm run clean && npm install` the check-coverage output show 100% coverage everywhere (This is off the `master` branch with no modifications and running on my Arch Linux laptop updated recently):
```
wink@wink-envy:~/prgs/alsatian.myfork (master)
$ npm run test

> alsatian@1.3.0 pretest /home/wink/prgs/alsatian.myfork
> npm run build


> alsatian@1.3.0 build /home/wink/prgs/alsatian.myfork
> tsc

...

> alsatian@1.3.0 unit-tests /home/wink/prgs/alsatian.myfork
> node ./cli/alsatian-cli.js "./test/unit-tests/**/*.spec.js"
TestSetConstructorTests
noTestsAtLocationGivesNoTestFixtures
|====================|

Pass: 3278/3278
Fail: 0/3278
Ignore: 0/3278

> alsatian@1.3.0 output-test-coverage /home/wink/prgs/alsatian.myfork
> nyc report

--------------------------------------|----------|----------|----------|----------|----------------|
File                                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------------------------------|----------|----------|----------|----------|----------------|
All files                             |      100 |      100 |      100 |      100 |                |
 cli                                  |      100 |      100 |      100 |      100 |                |
  alsatian-cli-options.ts             |      100 |      100 |      100 |      100 |                |
  alsatian-cli.ts                     |      100 |      100 |      100 |      100 |                |
  cli-test-runner.ts                  |      100 |      100 |      100 |      100 |                |
...
```
But if I do a second do another `npm run test` or directly run `npm run check-coverage` after the first successful test run the coverage report is for `.js` files instead of `.ts` files and does not show 100%:
```
wink@wink-envy:~/prgs/alsatian.myfork (master)
$ npm run check-coverage

> alsatian@1.3.0 check-coverage /home/wink/prgs/alsatian.myfork
> npm run build & nyc --reporter=lcov --reporter=html npm run unit-tests && npm run output-test-coverage


> alsatian@1.3.0 build /home/wink/prgs/alsatian.myfork
> tsc


> alsatian@1.3.0 unit-tests /home/wink/prgs/alsatian.myfork
> node ./cli/alsatian-cli.js "./test/unit-tests/**/*.spec.js"
TestSetConstructorTests
noTestsAtLocationGivesNoTestFixtures
|====================|

Pass: 3278/3278
Fail: 0/3278
Ignore: 0/3278

> alsatian@1.3.0 output-test-coverage /home/wink/prgs/alsatian.myfork
> nyc report

--------------------------------------|----------|----------|----------|----------|----------------|
File                                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------------------------------|----------|----------|----------|----------|----------------|
All files                             |    94.09 |    85.84 |    90.91 |    97.65 |                |
 cli                                  |    90.86 |    86.24 |     97.5 |    98.73 |                |
  alsatian-cli-options.js             |      100 |      100 |      100 |      100 |                |
  alsatian-cli.js                     |      100 |      100 |      100 |      100 |                |
  cli-test-runner.js                  |    83.18 |       80 |    94.74 |    97.14 |          22,23 |
 cli/errors                           |    79.49 |    55.56 |    72.41 |    88.57 |                |
  duplicate-cli-argument-error.js     |    76.47 |    53.85 |    71.43 |    86.67 |            4,5 |
  invalid-argument-names-error.js     |    82.61 |       60 |       75 |    90.48 |            4,5 |
...
```
What am I doing wrong?

**Comment by Jameskmonger** [comment]
You should be able to look at the generated html and see exactly what's going on.

**Comment by winksaville** [comment]
I looked at the html and it seems to me the problem is that for some reason nyc thinks on the second run its looking at it as a javascript app where as on the first run it knows its a typescript project. I assume you guys are not seeing the same problem, could you please test and make sure and then tell me what versions of `node`, `nyc` and `typescript` you're running:
```
$ node --version
v7.10.0
wink@wink-envy:~/foss/alsatian (master)
$ ./node_modules/.bin/nyc --version
10.3.0
wink@wink-envy:~/foss/alsatian (master)
$ ./node_modules/.bin/tsc --version
Version 2.3.2
wink@wink-envy:~/foss/alsatian (master)
```
In the mean time I'll dig deeper.

**Comment by jamesrichford** [comment]
Not seeing anything on my end.

A reason that this could be is the source maps are removed or not used on the second run. NYC will always treat the tests as JavaScript but checks for source maps when producing reports.

**Comment by winksaville** [comment]
Yeh, I'm thinking its something to do with the source maps too.

What do you mean not seeing anything?

**Comment by jamesrichford** [comment]
Haha sorry @winksaville I realise that was quite ambiguous. I meant that it seems to be working as expected when I checked but I only did a quick check.

**Comment by winksaville** [comment]
Hmmmm, I had a project I created which is using alsatian, nyc, tsc and guess what it was working fine with version `nyc 10.2.0` and `tsc 2.2.2`. I did a `yarn upgarde` and now that project is running `nyc 10.3.0` and `tsc 2.3.2` and guess what, the second run now reports the js coverage not the ts coverage!!

So I'll now revert each one back a version and see which is the culprit.

**Comment by winksaville** [comment]
So the culprit is `nyc 10.3.0` running it with either `tsc 2.2.2` or  `tsc 2.3.2` causes the second run to report number using `js` using `nyc 10.2.0` the report is always against the `ts` sources. I'll file a bug on nyc.

Could you validate what I've found and if so I think we should change `package.json` to set the devDependencies for  nyc version to 10.2.0.

**Comment by winksaville** [comment]
Culprit is `nyc 10.3.0` I've [file bug 565](https://github.com/istanbuljs/nyc/issues/565)

**Comment by winksaville** [comment]
Closing nyc is fixed 10.3.1 with [this PR](https://github.com/istanbuljs/nyc/pull/566)

**Comment by jamesrichford** [comment]
Awesome work @winksaville open source hero! :)
